File: src/Domain/Mobile/VoucherPin.cs
using Kogan.Mobile.Domain.Mobile.Enums;

namespace Kogan.Mobile.Domain.Mobile
{
    public sealed class VoucherPin
    {
        public int IdBatch { get; set; }

        public int IdVoucher { get; set; }

        public BatchVoucherAssociation BatchVoucherAssociation { get; set; }

        public string PinNumber { get; set; }

        public string Msisdn { get; set; }

        public VoucherPinStateEnum State { get; set; }

        public bool IsSold { get; set; } = false;

        public bool IsRedeemed { get; set; } = false;

        public bool IsExpired { get; set; } = false;
    }
}

File: src/Web/koganmobile-sapui5-app/webapp/models/VoucherPin.ts
import { VoucherPinStateEnum } from "../enums/VoucherPinStateEnum";

export default class VoucherPin {
    pinNumber: string;
    msisdn: string;
    state: VoucherPinStateEnum = VoucherPinStateEnum.None;
    isSold: boolean = false;
    isRedeemed: boolean = false;
    isExpired: boolean = false;
}